src/Neo/SmartContract/Native/LedgerContract.cs

```diff
@@ -47,13 +47,25 @@ internal override ContractTask OnPersist(ApplicationEngine engine)
             engine.Snapshot.Add(CreateStorageKey(Prefix_Block).Add(engine.PersistingBlock.Hash), new StorageItem(Trim(engine.PersistingBlock).ToArray()));
             foreach (TransactionState tx in transactions)
             {
-                engine.Snapshot.Add(CreateStorageKey(Prefix_Transaction).Add(tx.Transaction.Hash), new StorageItem(tx));
+                // It's possible that there are previously saved malicious conflict records for this transaction.
+                // If so, then remove it and store the relevant transaction itself.
+                var txKey = CreateStorageKey(Prefix_Transaction).Add(tx.Transaction.Hash);
+                engine.Snapshot.Delete(txKey);
+                engine.Snapshot.Add(txKey, new StorageItem(tx));
```

Comment:
@shargon or @Liaojinghui, I need your advice on this. What is the common way to rewrite an existing storage entity? 

`engine.Snapshot.Add` doesn't allow to rewrite entity if it's already in the storage. And what I need here is to put `tx` in the storage if it's not exist by the given key *or* rewrite an existing entity from whatever to `tx`.

Reply:
GetAndChange

Reply:
Let's try with it.